src/ast/codegen_llvm.cpp

```diff
@@ -962,6 +962,36 @@ void CodegenLLVM::visit(If &if_block)
   }
 }
 
+void CodegenLLVM::visit(Unroll &unroll)
+{
+  Function *parent = b_.GetInsertBlock()->getParent();
+  BasicBlock *loop = BasicBlock::Create(module_->getContext(), "loop", parent);
+  BasicBlock *done = BasicBlock::Create(module_->getContext(), "done", parent);
+
+  AllocaInst *val = b_.CreateAllocaBPF(SizedType(Type::integer, 8), "loop_count");
+  b_.CreateStore(b_.getInt64(unroll.var), val);
+
+  expr_ = val;
```

Comment:
expr_ is just intended for passing expressions back up the tree. You could just use val in the comparison below